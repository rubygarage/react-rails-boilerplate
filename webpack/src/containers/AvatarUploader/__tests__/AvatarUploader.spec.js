import React from 'react';
import AvatarUploaderContainer from 'containers/AvatarUploader';
import { shallow } from 'enzyme';

describe('UserEdit container', () => {
  const mockOnChange = jest.fn();
  const defaultProps = {
    input: {
      onChange: mockOnChange,
    },
  };

  const container = shallow(<AvatarUploaderContainer {...defaultProps} />);

  it('renders AvatarUploader component', () => {
    expect(container).toMatchSnapshot();
  });

  it('passes acepted file to input.onChange on drop', () => {
    const accepted = ['acceptedFile'];
    container.props().fileDropHandler(accepted);
    expect(mockOnChange.mock.calls[0][0]).toEqual(accepted);
  });

  it('passes empty string to input.onChange on drop', () => {
    container.props().handleRemoveSelectedFile();
    expect(mockOnChange.mock.calls[1][0]).toEqual('');
  });
});
