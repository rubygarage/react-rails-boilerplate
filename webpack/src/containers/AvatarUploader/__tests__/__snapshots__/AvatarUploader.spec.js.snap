// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEdit container renders AvatarUploader component 1`] = `
<AvatarUploader
  fileDropHandler={[Function]}
  handleRemoveSelectedFile={[Function]}
  input={
    Object {
      "onChange": [Function],
    }
  }
/>
`;
